src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -66,6 +68,7 @@ public S3BucketPublisher(String profileName, List<Entry> entries, List<MetadataP
         this.userMetadata = userMetadata;
 
         this.dontWaitForConcurrentBuildCompletion = dontWaitForConcurrentBuildCompletion;
+        this.consoleLogLevel = Level.parse(consoleLogLevel);
```

Comment:
Unfortunately, `parse` is executing under `synchronized`. In highly loaded environment it's not very efficient (we run ~1.5k jobs in parallel). I will think how to do it better.